### PR TITLE
Feature/adding local storage

### PR DESCRIPTION
**Pull request by jocelynjeffrey**

### CODE REVIEW

File: src/components/ProductList/ProductList.js [hunk 1]
```diff
@@ -4,15 +4,25 @@ import Product from '../Product';
...
-    );
+    props.productList.length > 0
+    ? products = props.productList.map(product => <Product
+      key={product.id}
+      {...product}
+      addToWishList={props.addToWishList}
+      removeFromWishList={props.removeFromWishList}
+      wishList={props.wishList} />)
+    : products = <h2>No results.</h2>
+
```
Comment (astr0-4) [inline src/components/ProductList/ProductList.js]: I think another way you could write this is:
`products = props.productList.length > 0 ? (1st case) : <h2>No results.</h2>`
It's slightly less code.

File: src/components/WishList/WishList.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  } else {
+    props.wishList.length > 0
+    ? products = props.wishList.map(product => <Product
+      key={product.id}
+      {...product}
+      addToWishList={props.addToWishList}
+      removeFromWishList={props.removeFromWishList}
+      wishList={props.wishList} />)
+    : products = <h2>No results.</h2>
+
```
Comment (astr0-4) [inline src/components/WishList/WishList.js]: same here

**Review by astr0-4 — APPROVED** [review]
Nice!! Just two small suggestions you can change if you want. 
Love the kittens lol 🐱

File: src/components/ProductList/ProductList.js (hunk 1 above)
Comment (jocelynjeffrey) [reply src/components/ProductList/ProductList.js]: updated!

File: src/components/WishList/WishList.js (hunk 2 above)
Comment (jocelynjeffrey) [reply src/components/WishList/WishList.js]: updated!

**Comment by astr0-4** [comment]
@jocelynjeffrey LGTM!

File: src/components/Product/Product.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+    <img src={product.image_thumb_url} alt="" />
+    <h2>{product.name}</h2>
     <p>
-      { (props.price_in_cents / 100).toFixed(2) }
-      <span>{props.package}</span>
+      { (product.price_in_cents / 100).toFixed(2) }
+      <span>{product.package}</span>
     </p>
   </li>
-);
```
Comment (jprittie) [inline src/components/Product/Product.js:12]: Should we include an "Add" button for each product?

File: src/redux/reducers/product.reducer.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    case PRODUCT_ACTIONS.REMOVE_PRODUCT_FROM_WISHLIST:
+      return {...state,
+        wishList: state.wishList
+        .filter(product => product.id !== action.payload.id)
+      };
+
+    default:
+      return state;
+  }
+}
```
Comment (jprittie) [inline src/redux/reducers/product.reducer.js:48]: I think that once we have live data we should separate out this into two reducers... one to handle adding and removing from the wishlist, and another to handle the search results we get back from the epic.

**Review by jprittie — APPROVED** [review]
Hello... the first kitten is definitely a better deal!

Adding to/removing from the wishlist is working well, as is syncing with localStorage... all of this seems good to merge. I left a comment about the product reducer... I think we should consider splitting it up into two reducers once we have live data.

File: src/redux/reducers/product.reducer.js (hunk 4 above)
Comment (jocelynjeffrey) [reply src/redux/reducers/product.reducer.js:48]: Yeah, once we get live data we should definitely take a look at some of our structures - maybe we can make refactoring a trello card or task.

File: src/components/Product/Product.js (hunk 3 above)
Comment (jocelynjeffrey) [reply src/components/Product/Product.js:12]: Yeah! For right now I made the entire product into a pseudo button to skip styling, but once we figure out what we want the add/remove product process to look like, we should probably add a button here.

**Pull request merged**